openedx_external_enrollments/tests/external_enrollments/tests_pathstream_external_enrollments.py

```diff
@@ -29,7 +33,332 @@ def test_str(self):
+            },
+        }
+        log = 'File content [{}] was downloaded from S3 for [{}]'.format(s3_file, self.base.__str__())
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            file_content = self.base._download_file()  # pylint: disable=protected-access
+
+            log_capture.check(
+                (self.module, 'INFO', log),
+            )
+
+        self.base.client.get_object.assert_called_with(
+            Bucket=s3_bucket,
+            Key=s3_file,
+        )
+        self.assertEqual(file_content, expected_file_content)
+
+    def test_download_file_error(self):
+        """If no connection, credentials or wrong parameter values
+        _download_file should handle ClientError and raise PathstreamTaskExecutionError."""
+        code = '400'
+        message = 'Failed'
+        operation = 'get_object'
+        attrs = {
+            'get_object.side_effect': ClientError(
+                {
+                    'Error': {
+                        'Code': code,
+                        'Message': message,
+                    },
+                },
+                operation,
+            ),
+        }
+        clienterror_msg = 'An error occurred ({}) when calling the {} operation: {}'.format(
+            code,
+            operation,
+            message,
+        )
+        self.base.client = Mock(spec=boto3.client('s3'), **attrs)
+
+        with self.assertRaisesMessage(PathstreamTaskExecutionError, clienterror_msg):
+            self.base._download_file()  # pylint: disable=protected-access
+
+    def test_prepare_new_content(self):
+        """This test validates that _prepare_new_content accomplish
+        its task of sorting and encoding."""
+        content = [
+            'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n',
+            'course_T2,2@mail,2021-06-29 16:51:00.456900,true\n',
+            'course_T1,1@mail,2021-06-29 16:50:00.456900,true\n',
+            'course_T2,2@mail,2021-06-29 16:52:00.456900,false\n',
+        ]
+        expected_result = (
+            b'course_T1,1@mail,2021-06-29 16:50:00.456900,true\n'
+            b'course_T2,2@mail,2021-06-29 16:51:00.456900,true\n'
+            b'course_T2,2@mail,2021-06-29 16:52:00.456900,false\n'
+            b'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n'
+        )
+
+        result = self.base._prepare_new_content(content)  # pylint: disable=protected-access
+
+        self.assertEqual(result, expected_result)
+
+    def test_prepare_new_content_error(self):
+        """This test validates that _prepare_new_content raises
+        PathstreamTaskExecutionError when called with invalid data."""
+        content = [
+            'course_T1,1@mail,2021-06-29 16:53:00.456900,false\n',
+            'invalid_data\n',
+        ]
+
+        with self.assertRaises(PathstreamTaskExecutionError):
+            self.base._prepare_new_content(content)  # pylint: disable=protected-access
+
+    def test_upload_file(self):
```

Comment:
You can also check that 'hashlib.md5' and 'base64.b64encode' have been called. @Jacatove